common/templates/context_funcs.go

```diff
@@ -1086,6 +1086,53 @@ func (c *Context) tmplGetMessage(channel, msgID interface{}) (*discordgo.Message
 	return message, nil
 }
 
+const (
+	fetchBatchSize            = 100                 // max count Discord API supports fetching in one call
+	maxSupportedReactionCount = fetchBatchSize * 50 // 50 API calls
+)
+
+func (c *Context) tmplGetAllMessageReactions(channel, msgID, emoji interface{}) (interface{}, error) {
+	if c.IncreaseCheckGenericAPICall() {
+		return nil, ErrTooManyAPICalls
+	}
+
+	if c.IncreaseCheckCallCounter("get_all_message_reactions", 1) {
+		return nil, errors.New("can call getAllMessageReactions at max once")
+	}
+
+	cID := c.ChannelArgNoDM(channel)
+	if cID == 0 {
+		return nil, nil
+	}
+
+	mID := ToInt64(msgID)
+	e := ToString(emoji)
+
+	var (
+		users []*discordgo.User
+		after int64 = 0
+	)
+
+	// Discord API limits amount to fetch per call to 100; there may be more reactions than that, so loop to get all
+	for {
+		received, err := common.BotSession.MessageReactions(cID, mID, e, fetchBatchSize, 0, after)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(received)+len(users) > maxSupportedReactionCount {
```

Comment:
Perhaps here instead of returning `nil` and an `error`, it could return what has been fetched already. What do you think?

Reply:
From the user's perspective, it is not possible to know whether the received output is complete or incomplete in such a case. Consider the common use-case of a giveaway command -- it would be preferable to have it error if there are too many reactions instead of silently skipping over some users.

Reply:
The function is hard limited to 5k users, right?

If it fetched more than 5k, why not simply return the first 5k?
It should be well documented for users that this function is hard capped at 5k. Thus if they use it in a reaction with more than 5k users, they should know that they will get the first 5k users that reacted, not everyone. 

Otherwise it is best to check if the reaction has more than 5k reactors and just simply return an error, otherwise we will be using 50 API Calls for "nothing".

Let me know what you think.

To be honest, I think exposing `after` and `before` to the users would be best, perhaps they could be optional arguments. But I can understand the benefit of "hiding" them.

Reply:
I see your point but again I'm not too convinced that returning partial data is the way to go -- it's much easier for users to introduce subtle bugs.

> To be honest, I think exposing after and before to the users would be best, perhaps they could be optional arguments. But I can understand the benefit of "hiding" them.

I disagree; I can't see any use case for fetching part of the reactions -- only all of them. (Hence why I'm against returning partial data.)